pkg/envtest/server.go

```diff
@@ -135,10 +135,10 @@ type Environment struct {
 }
 
 // Stop stops a running server.
-// If USE_EXISTING_CLUSTER is set to true, this method is a no-op.
+// If USE_EXISTING_CLUSTER is set to true, uninstall whatever CRDs were installed with Start().
 func (te *Environment) Stop() error {
 	if te.useExistingCluster() {
-		return nil
+		return UninstallCRDs(te.Config, te.CRDInstallOptions)
```

Comment:
I don't think we should be doing this by default because:

* This may be unexpected for users that re-use an existing cluster they have workloads on that use the given CRD they care
* This breaks parallel tests that use the same cluster

Maybe make this an opt-in option of the `Environment`?

Reply:
This makes total sense! I'm _ashamed_ it didn't cross my mind. Will do.